pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -478,7 +478,7 @@ private CompletableFuture<Void> individualAckWithTransaction(CommandAck ack) {
             MessageIdData msgId = ack.getMessageIdAt(i);
             PositionImpl position;
             long ackedCount = 0;
-            long batchSize = getBatchSize(msgId);
+            long batchSize = msgId.getBatchSize();
```

Comment:
don't change this method; if this message is not a batch message, the batchSize will set it to -1. Batch size is used to change unAckMessageCount, is it should be 1, not -1, and not 0. so I only suggest change positionsAcked not this batchSize